packages/Meshfree/test/tstNearestNeighborExodusGenerator.cpp

```diff
@@ -0,0 +1,259 @@
+/****************************************************************************
+ * Copyright (c) 2012-2017 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ ****************************************************************************/
+
+#include "DTK_ConfigDefs.hpp"
+#include "DTK_Types.h"
+
+#include "PointCloudProblemGenerator/ExodusProblemGenerator.hpp"
+#include "PointCloudProblemGenerator/PointCloudProblemGenerator.hpp"
+#include <DTK_DetailsDistributedSearchTreeImpl.hpp>
+#include <DTK_NearestNeighborOperator.hpp>
+
+#include <Kokkos_View.hpp>
+
+#include <Teuchos_Array.hpp>
+#include <Teuchos_CommHelpers.hpp>
+#include <Teuchos_DefaultComm.hpp>
+#include <Teuchos_RCP.hpp>
+#include <Teuchos_UnitTestHarness.hpp>
+
+#include <cmath>
+#include <limits>
+#include <memory>
+#include <string>
+
+//---------------------------------------------------------------------------//
+// Brute-force calculate the nearest neighbors
+template <class Device>
+void computeNeighborsBruteForce(
+    const Teuchos::RCP<const Teuchos::Comm<int>> &comm,
+    const std::string &src_exodus_file, const std::string &tgt_exodus_file,
+    const Kokkos::View<DataTransferKit::Coordinate **, Kokkos::LayoutLeft,
+                       Device> &tgt_coords,
+    Kokkos::View<DataTransferKit::Coordinate **, Kokkos::LayoutLeft,
+                 Kokkos::Serial> &src_nearest_coords )
```

Comment:
Why do you work from file names rather than from `src_coords` and `tgt_coords`?

Reply:
To make it easy I just wanted to reload the coordinates in a serial partitioning on each rank. This obviously means doing some extra work but it was an easy way to get the coordinates.

Reply:
I do not see how it is easier than passing the two point cloud since you have already parsed the files and then possibly use Tpetra import/export capability if you want to import the whole source vector on the current rank.

Reply:
Right - I can refactor to do the import/export